boa/src/exec/tests.rs

```diff
@@ -912,3 +912,23 @@ fn to_string() {
     assert_eq!(engine.to_string(&Value::rational(55.0)).unwrap(), "55");
     assert_eq!(engine.to_string(&Value::string("hello")).unwrap(), "hello");
 }
+
+#[test]
+fn calling_function_with_unspecified_arguments() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+    let scenario = r#"
+        function fibonacci(num, memo) {
```

Comment:
I think this test is too broad. We are not only checking to see the behaviour of unspecified arguments, but also array access, || operator, recursion and more. Maybe we should do a check with a simpler method, such as
```js
function test(a, b) {
    return b;
}

test(10)
```
``` assert_eq!(foward(&mut engine, scenario), "undefined"); ```

Or something similar.

Reply:
Yes. this is probably the best option. I just took the code that was broken from the issue as the test.